tests/xharness/Logging/CaptureLog.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.IO;
+
+namespace Xharness.Logging {
+
+	// A log that captures data written to a separate file between two moments in time
+	// (between StartCapture and StopCapture).
+	public class CaptureLog : Log
+	{
+		public string CapturePath { get; private set; }
+		public string Path { get; private set; }
+
+		long startPosition;
+		long endPosition;
+		bool entire_file;
+		bool started;
+
+		public CaptureLog (ILogs logs, string path, string capture_path, bool entire_file = false)
+			: base (logs)
+		{
+			if (path == null)
+				throw new ArgumentNullException (nameof (path));
+
+			Path = path;
+			CapturePath = capture_path;
+			this.entire_file = entire_file;
+		}
+
+		public void StartCapture ()
+		{
+			if (entire_file)
+				return;
+
+			if (File.Exists (CapturePath))
+				startPosition = new FileInfo (CapturePath).Length;
+			started = true;
+		}
+
+		public void StopCapture ()
+		{
+			if (!started && !entire_file)
+				throw new InvalidOperationException ("StartCapture most be called before StopCature on when the entire file will be captured.");
+			if (!File.Exists (CapturePath)) {
+				File.WriteAllText (Path, $"Could not capture the file '{CapturePath}' because it doesn't exist.");
+				return;
+			}
+
+			if (entire_file) {
+				File.Copy (CapturePath, Path, true);
+				return;
+			}
+
+			endPosition = new FileInfo (CapturePath).Length;
+
+			Capture ();
+		}
+
+		void Capture ()
```

Comment:
```suggestion
		private void Capture ()
```

But I would also maybe rename this to `Finalize` or `Flush` or something..

Reply:
private is the default in c#. Because it is, we do not add it since in the mono style it is not recommended because it will 'clutter' the code.

Reply:
`internal` is the default

Reply:
From the docs:

> Class members, including nested classes and structs, can be public, protected internal, protected, internal, private protected, or private. Class and struct members, including nested classes and structs, have private access by default.

[Source](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/access-modifiers)

Since this is a member, not a class, it is private by default. Also, [the mono style guide](https://www.mono-project.com/community/contributing/coding-guidelines/#baroque-coding)

Reply:
Oops, my bad, had no idea it's different for members

Reply:
No problem :)